MinimizeIPTablesRestore to GA

danwinship opened the pull request:
#### What type of PR is this?
/kind feature
/sig network
/priority important-soon

#### What this PR does / why we need it:
Move KEP-3453 `MinimizeIPTablesRestore` to GA

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

(right? since it was already enabled in 1.27?)

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3453
```

danwinship committed:
MinimizeIPTablesRestore to GA

aojea commented:
metrics merged https://github.com/kubernetes/kubernetes/pull/117787

/lgtm
/approve
/assign @thockin

wojtek-t commented:
/lgtm
/approve

/hold
In case you want @thockin to take a look (but given he approved the KEP, feel free to hold cancel yourself).

danwinship commented:
Removed the feature gate checks so someone won't have to deal with that later.
(Also added a comment since otherwise it looked like `tryPartialSync` could be optimized out, but it can't be.)

wojtek-t commented:
/lgtm

/hold cancel
